quattor/types/metadata.pan

```diff
@@ -14,7 +14,28 @@ type structure_template = {
     "branch" : structure_branch
 };
 
+type structure_os_version = {
+    @{Descriptive form of version, e.g. 7x}
+    "name" : string
+    @{Major version number component, e.g. 7}
+    "major" : long(0..)
+    @{Minor version number component}
+    "minor" ? long(0..)
+};
+
+type structure_os = {
+    @{Distribution name, e.g. sl}
+    "distribution" : string
```

Comment:
i'm not a big fan of `string` without some basic checking. this is to avoid spelling issues, abreviations etc. there aren't that many we need to add here. i would make a distribution type with official names and also it for the `family` attribute.
is there a reason why it has to be `sl` and not `Scientific Linux`?